source/material_model/rheology/elasticity.cc

```diff
@@ -0,0 +1,381 @@
+
+        use_stress_averaging = prm.get_bool ("Use stress averaging");
+        if (use_stress_averaging)
+          AssertThrow(use_fixed_elastic_time_step == true,
+                      ExcMessage("Stress averaging can only be used if 'Use fixed elastic time step' is set to true'"));
+
+        fixed_elastic_time_step = prm.get_double ("Fixed elastic time step");
+        AssertThrow(fixed_elastic_time_step > 0,
+                    ExcMessage("The fixed elastic time step must be greater than zero"));
+
+        if (this->convert_output_to_years())
+          fixed_elastic_time_step *= year_in_seconds;
+
+        AssertThrow(this->get_parameters().enable_elasticity == true,
+                    ExcMessage ("Material model Viscoelastic only works if 'Enable elasticity' is set to true"));
+
+        // Check whether the compositional fields representing the viscoelastic
+        // stress tensor are both named correctly and listed in the right order.
+        AssertThrow(this->introspection().compositional_index_for_name("stress_xx") == 0,
+                    ExcMessage("Rheology model Elasticity only works if the first "
+                               "compositional field is called stress_xx."));
+        AssertThrow(this->introspection().compositional_index_for_name("stress_yy") == 1,
+                    ExcMessage("Rheology model Elasticity only works if the second "
+                               "compositional field is called stress_yy."));
+        if (dim == 2)
+          {
+            AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 2,
+                        ExcMessage("Rheology model Elasticity only works if the third "
+                                   "compositional field is called stress_xy."));
+          }
+        else if (dim == 3)
+          {
+            AssertThrow(this->introspection().compositional_index_for_name("stress_zz") == 2,
+                        ExcMessage("Rheology model Elasticity only works if the third "
+                                   "compositional field is called stress_zz."));
+            AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 3,
+                        ExcMessage("Rheology model Elasticity only works if the fourth "
+                                   "compositional field is called stress_xy."));
+            AssertThrow(this->introspection().compositional_index_for_name("stress_xz") == 4,
+                        ExcMessage("Rheology model Elasticity only works if the fifth "
+                                   "compositional field is called stress_xz."));
+            AssertThrow(this->introspection().compositional_index_for_name("stress_yz") == 5,
+                        ExcMessage("Rheology model Elasticity only works if the sixth "
+                                   "compositional field is called stress_yz."));
+          }
+        else
+          ExcNotImplemented();
```

Comment:
that doesn't do what you want without the assert. ;-)

Reply:
Wow, I didn't even see that...